src/components/application_manager/src/commands/mobile/create_interaction_choice_set_request.cc

```diff
@@ -151,8 +151,13 @@ mobile_apis::Result::eType CreateInteractionChoiceSetRequest::CheckChoiceSet(
     for (next_choice_set_it = current_choice_set_it + 1;
          choice_set->end() != next_choice_set_it;
          ++next_choice_set_it) {
-      if (compareSynonyms(*current_choice_set_it, *next_choice_set_it)) {
-        return mobile_apis::Result::DUPLICATE_NAME;
+      const bool menu_name_duplicated =
+          compareStr((*current_choice_set_it)[strings::menu_name],
+                     (*next_choice_set_it)[strings::menu_name]);
+      const bool choice_set_duplicated =
+          compareSynonyms(*current_choice_set_it, *next_choice_set_it);
```

Comment:
@okozlovlux, current_choice_set must be different, menu_name also must be different.
In the both cases we return mobile_api::Result::DUPLICATE_NAME.